src/api/package/metadata.ts

```diff
@@ -3,11 +3,24 @@ import { API_PATHS } from "../../constants/url";
 import { PackageTagConfig } from "../config";
 import { getAssetsPath } from "./util";
 
+export interface ConstructFramework {
+  name: CDKType;
+  majorVersion?: number;
+}
+
 export interface Metadata {
-  constructFramework?: {
-    name?: CDKType;
-    majorVersion?: number;
-  };
+  /**
+   * Describes the associated Construct framework for a library.
+   * Back-end will introduce a new metadata.constructFrameworks property to replace metadata.constructFramework
+   * @deprecated Use constructFrameworks instead
+   */
+  constructFramework?: Partial<ConstructFramework>;
```

Comment:
Why is this partial?

Reply:
Just so that I could re-use the type for the new implementation. The old signature allowed `cdkType` and `majorVersion` to be undefined, whereas moving forward we should only define a constructFramework object if there is a `cdkType` associated